Installation issue: py-torch +breakpad fails to build on RHEL@9.2

### Steps to reproduce the issue

```console
$ spack install py-torch@1.10%gcc=7.5.0 arch=x86_64
```


### Error message

<details><summary>Error message</summary>

<pre>
  >> 3433    /tmp/sbulut/spack-stage/spack-stage-py-torch-1.10.0-56lrpr454rywkayd2smjuvrtjuvqssa3/spack-src/third_par
             ty/breakpad/src/client/linux/handler/exception_handler.cc:141:65: error: no matching function for call t
             o 'max(int, long int)'
     3434       static const unsigned kSigStackSize = std::max(16384, SIGSTKSZ);
     3435                                                                     ^
     3436    In file included from /disk/hpcpm/3vg_spack/0171/opt/spack/linux-rhel9-x86_64/gcc-11.3.1/gcc-7.5.0-ldgnc
             vpwbmnr3kjxaxdzarrvecdyl5ov/include/c++/7.5.0/bits/char_traits.h:39:0,
     3437                     from /disk/hpcpm/3vg_spack/0171/opt/spack/linux-rhel9-x86_64/gcc-11.3.1/gcc-7.5.0-ldgnc
             vpwbmnr3kjxaxdzarrvecdyl5ov/include/c++/7.5.0/string:40,
     3438                     from /tmp/sbulut/spack-stage/spack-stage-py-torch-1.10.0-56lrpr454rywkayd2smjuvrtjuvqss
             a3/spack-src/third_party/breakpad/src/client/linux/handler/exception_handler.h:38,
     3439                     from /tmp/sbulut/spack-stage/spack-stage-py-torch-1.10.0-56lrpr454rywkayd2smjuvrtjuvqss
             a3/spack-src/third_party/breakpad/src/client/linux/handler/exception_handler.cc:66:

</pre></details>


### Information on your system

$ spack debug report
* **Spack:** 0.17.1
* **Python:** 3.9.16
* **Platform:** linux-rhel9-skylake_avx512
* **Concretizer:** clingo


### Additional information

can provide build out later if necessary.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/py-torch/package.py
+++ b/var/spack/repos/builtin/packages/py-torch/package.py
@@ -103,7 +103,7 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         "breakpad",
         default=True,
         description="Enable breakpad crash dump library",
-        when="@1.9:1.11",
+        when="@1.10:1.11",
     )
 
     conflicts("+cuda+rocm")
@@ -284,6 +284,14 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         "https://github.com/pytorch/pytorch/pull/55063.patch?full_index=1",
         sha256="2229bcbf20fbe88aa9f7318f89c126ec7f527875ffe689a763c78abfa127a65c",
         when="@1.1:1.8.1",
+    )
+
+    # https://github.com/pytorch/pytorch/issues/70297
+    patch(
+        "https://github.com/google/breakpad/commit/605c51ed96ad44b34c457bbca320e74e194c317e.patch?full_index=1",
+        sha256="694d83db3a2147d543357f22ba5c8d5683d0ed43e693d42bca8f24ec50080f98",
+        when="+breakpad",
+        working_dir="third_party/breakpad",
     )
 
     # Fixes CMake configuration error when XNNPACK is disabled

